File: 프로그래머스/1/388351. 유연근무제/유연근무제.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int schedules[], size_t schedules_len, int** timelogs, size_t timelogs_rows, size_t timelogs_cols, int startday) {
    int answer = 0;
    
    for (size_t i = 0; i < schedules_len; i++) {
        int deadline = (schedules[i] / 100 * 60) + (schedules[i] % 100) + 10;
        bool is_eligible = true;
        
        for (size_t j = 0; j < timelogs_cols; j++) {
            int current_day = (startday + (int)j - 1) % 7 + 1;
            
            if (current_day == 6 || current_day == 7) {
                continue;
            }
            
            int arrival_time = (timelogs[i][j] / 100 * 60) + (timelogs[i][j] % 100);
            
            if (arrival_time > deadline) {
                is_eligible = false;
                break;
            }
        }
        
        if (is_eligible) {
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/1/388351. 유연근무제/유연근무제.py
def solution(schedules, timelogs, startday):
    answer = 0
    for employ, time in zip(schedules, timelogs):
        deadline = (employ // 100 * 60) + (employ % 100) + 10
        is_TF = 1
        day = startday
        for i in time:
            intime = (i // 100 * 60) + (i % 100)
            if day >= 8:
                day %= 7
            if day == 6 or day == 7:
                day += 1
                continue
            
            if intime <= deadline:
                day += 1
                continue
            else:
                is_TF = 0
                break
        
        if is_TF:
            answer += 1
    
    return answer